### PR TITLE
Replace got with built-in http/https node module

### PR DESCRIPTION
**Pull request by z4kn4fein**
### Describe the purpose of your pull request

This PR replaces `got` with a `http`/`https` node module based HTTP handler.

### Related issues (only if applicable)

#45 #33 

### Requirement checklist (only if applicable)

- [x] I have covered the applied changes with automated tests.
- [x] I have executed the full automated test set against my changes.
- [x] I have validated my changes against all supported platform versions.
- [x] I have read and accepted the [contribution agreement](https://github.com/configcat/legal/blob/main/contribution-agreement.md).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "configcat-node",
-  "version": "7.0.0",
```
Comment (sigewuzhere) [inline package.json:3]: This might be a major

**Review by sigewuzhere — COMMENTED** [review]
This might be a major version update.

File: package.json [hunk 2]
```diff
@@ -27,7 +27,6 @@
   "homepage": "https://configcat.com",
   "dependencies": {
     "configcat-common": "^6.0.0",
-    "got": "^9.6.0",
```
Comment (endret) [inline package.json:30]: <3

**Review by endret — APPROVED** [review]
good job!

File: package.json (hunk 1 above)
Comment (endret) [reply package.json:3]: Why do you think it? I can not see any non backward compatible changes.

Comment (sigewuzhere) [reply package.json:3]: Let's discuss on our next team meeting.

Comment (sigewuzhere) [reply package.json:3]: The team agreed on bumping the major.

**Review by sigewuzhere — APPROVED** [review]

**Pull request merged**